### PR TITLE
IR Drop Reports in the Signoff Stage

### LINKED ISSUE #1238: Increasing SYNTH_MAX_FANOUT results in crash during PDN
### Description

I was increasing SYNTH_MAX_FANOUT to try to remove the warning about fanout. At 15, the PDN fails and the flow stops with an error.

### Environment
```
python3 ./env.py issue-survey
Kernel: Linux v5.15.0-41-generic
Distribution: ubuntu 20.04
Python: v3.8.10 (OK)
Container Engine: docker v20.10.7 (OK)
OpenLane Git Version: f9b5781f5ef0bbdf39ab1c2bbd78be8db11b27f2
pip: INSTALLED
pip:venv: INSTALLED
---
PDK Version Verification Status: OK
---
Git Log (Last 3 Commits)

f9b5781 2022-07-01T16:04:31+02:00 Fix a bug with `-overwrite` (#1171) - Anton Blanchard -  (grafted, HEAD, tag: 2022.07.02_01.38.08)
```

### Reproduction Material

* Upload a tarball containing the relevant design. Have done. Didn't bother with issue_reproducible - see below. 

[RUN_2022.07.27_14.37.12.tar.gz](https://github.com/The-OpenROAD-Project/OpenLane/files/9200169/RUN_2022.07.27_14.37.12.tar.gz)

./flow -design seven_segment_seconds

### Expected behavior

Tools should finish the flow.

### Other questions

Why is issue_reproducible so huge? The uncompressed directory is 7.2GB. I had a quick look and it seems to recursively pack itself.
28k files. Compressed with tar.gz -> 1.2GB. Didn't bother attaching here.

### LINKED ISSUE #1241: "FAQs" link not available
### Prompt
The FAQs in the main readme file is not available.

### Proposal
Edit the link.

### LINKED ISSUE #1247: move analyze_power_grid  to the end of the flow
### Description
analyze_power_grid should be called at the end of the flow.  Currently it is called in pdn.tcl when no instances have been placed and generates meaningless results.

### Environment
Latest version

### Reproduction Material
Run any design, e.g. 'make test'

### Expected behavior
A meaningful IR drop report

### Logs
from 'make test'
```
[WARNING PSM-0071] Instance _506_ is not placed. Therefore, the power drawn by this instance is not considered for IR  drop estimation. Please run analyze_power_grid after instances are placed.
[INFO PSM-0064] Number of voltage sources = 1.
[INFO PSM-0040] All PDN stripes on net VPWR are connected.
########## IR report #################
Worstcase voltage: 1.80e+00 V
Average IR drop  : 1.04e-10 V
Worstcase IR drop: 2.59e-10 V
######################################
```

### PR DESCRIPTION
**Pull request by donn**
\+ Create new IR drop report generation step in the signoff stage
\+ ALL* STEPS NOW PRINT THEIR LOGS' PATH
~ `increment_index/TIMER/set log/puts_info` commands organized across the board
~ Fix issue where `FP_PDN_AUTO_ADJUST` would not snap its values to the mfg grid
~ Reorder `configuration/general.tcl`
~ Update Contribution Guidelines
~ Various Docs Fixes
\- Remove `FP_PDN_IRDROP`

\* All /ɔːl/ *noun*: used to refer to the whole quantity or extent of a particular group or thing so long as said thing does not include global routing or lvs as they are written quite poorly.

---

Resolves #1238
Resolves #1241
Resolves #1247

### CODE REVIEW

**Comment by maliberty** [comment]
You have a lot of changes in here that are unrelated to the title of this PR.  I don't intend to review them (lvs, klayout, magic, docs, etc).

File: scripts/openroad/irdrop.tcl [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    exit 1
+}
+
+if {[catch {read_def $::env(CURRENT_DEF)} errmsg]} {
+    puts stderr $errmsg
+    exit 1
+}
+
+source $::env(SCRIPTS_DIR)/openroad/set_rc.tcl
+analyze_power_grid -net $::env(VDD_NET) -outfile $::env(_tmp_save_rpt)
```
Comment (maliberty) [inline scripts/openroad/irdrop.tcl:26]: You need to load a Liberty or set the voltage per our discussion in your OR issue.

**Review by maliberty — CHANGES_REQUESTED** [review]

File: scripts/tcl_commands/floorplan.tcl [hunk 2]
```diff
@@ -157,7 +172,7 @@ proc place_io_ol {args} {
         --length $arg_values(-length)\
         {*}$flags_map(-unmatched_error)\
         {*}$arg_values(-extra_args)\
-        $arg_values(-def) |& tee [index_file $::env(floorplan_logs)/place_io_ol.log] $::env(TERMINAL_OUTPUT)
+        $arg_values(-def) |& tee  $::env(TERMINAL_OUTPUT)
```
Comment (kareefardi) [inline scripts/tcl_commands/floorplan.tcl:175]: i think you missed `$log` here

File: scripts/tcl_commands/placement.tcl [hunk 3]
```diff
@@ -48,14 +50,15 @@ proc global_placement {args} {
...
+    set log [index_file $::env(placement_logs)/global.log]
+    puts_info "Performing Random Global Placement (log: [relpath . $log])..."
     set ::env(SAVE_DEF) [index_file $::env(placement_tmpfiles)/global.def]
 
     try_catch $::env(OPENROAD_BIN) -python $::env(SCRIPTS_DIR)/odbpy/random_place.py\
         --output $::env(SAVE_DEF) \
         --input-lef $::env(MERGED_LEF) \
         $::env(CURRENT_DEF) \
-        |& tee $::env(TERMINAL_OUTPUT) [index_file $::env(placement_logs)/global.log]
+        |& tee $::env(TERMINAL_OUTPUT)
```
Comment (kareefardi) [inline scripts/tcl_commands/placement.tcl:61]: same here

File: scripts/tcl_commands/placement.tcl [hunk 4]
```diff
@@ -129,20 +134,21 @@ proc manual_macro_placement {args} {
 
     try_catch openroad -python\
         $::env(SCRIPTS_DIR)/odbpy/manual_macro_place.py {*}$arg_list |&\
-        tee $::env(TERMINAL_OUTPUT) [index_file $::env(placement_logs)/macro_placement.log]
+        tee $::env(TERMINAL_OUTPUT)
```
Comment (kareefardi) [inline scripts/tcl_commands/placement.tcl:137]: here as well

File: scripts/openroad/irdrop.tcl (hunk 1 above)
Comment (donn) [reply scripts/openroad/irdrop.tcl:37]: Done.

File: scripts/tcl_commands/floorplan.tcl (hunk 2 above)
Comment (donn) [reply scripts/tcl_commands/floorplan.tcl]: You're eagle-eyed, I did a scan for these and still ended up missing a couple. Thanks.

**Review by kareefardi — APPROVED** [review]

**Pull request merged**